### PR TITLE
cylc-insert: prevent insertion at invalid cycle point.

### LINKED ISSUE #1239: Disallow manual insertion at non-graphed cycle points
Currently (and somewhat surprisingly) tasks can be manually inserted at invalid cycle points (i.e. where they don't appear in the suite graph) but they end up with no prerequisites.  E.g. initial R1 tasks inserted at a cycle point beyond the initial one; or cycling tasks inserted off-sequence.

I'm pretty sure this was not intended; we should disallow it.

### PR DESCRIPTION
**Pull request by oliver-sanders**
Close #1239 

Don't insert a task unless its cycle point is on one of the tasks sequences. Over-ridable by the `--no-check` option.

### CODE REVIEW

File: lib/cylc/network/https/suite_command_server.py [hunk 1]
```diff
@@ -148,13 +148,14 @@ def nudge(self):
...
-    def insert_tasks(self, items, stop_point_string=None):
+    def insert_tasks(self, items, stop_point_string=None, no_check=False):
         if not isinstance(items, list):
             items = [items]
         if stop_point_string == "None":
             stop_point_string = None
         return self._put("insert_tasks", (items,),
-                         {"stop_point_string": stop_point_string})
+                         {"stop_point_string": stop_point_string,
+                          "no_check": True if no_check == 'True' else False})
```
Comment (matthewrmshin) [inline lib/cylc/network/https/suite_command_server.py]: Should be sufficient to do: `"no_check": no_check == 'True'`.

But then why would `no_check` be a string?

Comment (oliver-sanders) [inline lib/cylc/network/https/suite_command_server.py]: Objects serialised for transmission on the comms layer? See line 154 in the diff above where `stop_point_string` is treated in a similar way.

Comment (matthewrmshin) [inline lib/cylc/network/https/suite_command_server.py]: OK. I forgot it was done by `urllib.urlencode` not `json.dumps`.

**Comment by hjoliver** [comment]
(I guess you just rebased the branch; I'll wait on Travis CI before merging...)

**Comment by hjoliver** [comment]
Huh, Travis CI doesn't like `tests/restart/19-checkpoint.t`.  It passes in my environment (and I updated my remote branch after the rebase).

**Comment by matthewrmshin** [comment]
`tests/restart/19-checkpoint.t` has become a bit sensitive lately (mainly in my environment, but it looks like Travis CI is now affected). I don't think the failure is related to this change. I'll have a look at the test again in due course.

**Comment by oliver-sanders** [comment]
Passes in my environment.

**Comment by matthewrmshin** [comment]
(I have bundled a change in #2100 to improve the stability of `tests/restart/19-checkpoint.t`.)

**Comment by hjoliver** [comment]
In that case, merging now...

**Pull request merged**